Py_TRASHCAN_SAFE* was deprecated and removed in 3.13 with drop in replacement.
author: micketeer@gmail.com
--- blist/_blist.c.orig2	2025-04-30 16:32:07.809662996 +0000
+++ blist/_blist.c	2025-04-30 16:49:32.679953788 +0000
@@ -5798,7 +5798,7 @@
         if (_PyObject_GC_IS_TRACKED(self))
                 PyObject_GC_UnTrack(self);
 
-        Py_TRASHCAN_SAFE_BEGIN(self)
+        Py_TRASHCAN_BEGIN(self, py_blist_dealloc)
 
         /* Py_XDECREF() is needed here because the Python C API allows list
          * items to be NULL. */
@@ -5821,7 +5821,7 @@
                 Py_TYPE(self)->tp_free((PyObject *)self);
         }
 
-        Py_TRASHCAN_SAFE_END(self);
+        Py_TRASHCAN_END(self);
 }
 
 BLIST_PYAPI(int)
